x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/plan/logical/TopNSerializationTests.java

```diff
@@ -40,16 +41,26 @@ protected TopN mutateInstance(TopN instance) throws IOException {
         LogicalPlan child = instance.child();
         List<Order> order = instance.order();
         Expression limit = instance.limit();
-        switch (between(0, 2)) {
+        boolean local = instance.isLocal();
+        switch (between(0, 3)) {
             case 0 -> child = randomValueOtherThan(child, () -> randomChild(0));
             case 1 -> order = randomValueOtherThan(order, TopNSerializationTests::randomOrders);
             case 2 -> limit = randomValueOtherThan(limit, AbstractExpressionSerializationTests::randomChild);
+            case 3 -> local = local == false;
         }
-        return new TopN(source, child, order, limit);
+        return new TopN(source, child, order, limit, local);
     }
 
     @Override
     protected boolean alwaysEmptySource() {
         return true;
     }
+
+    @Override
+    protected TopN copyInstance(TopN instance, TransportVersion version) throws IOException {
+        // TopN#local is ALWAYS false after serialization.
+        TopN deserializedCopy = super.copyInstance(instance.withLocal(false), version);
```

Comment:
I don't think you need to set the original instances `local`, it'll be igored by the serialization, anyway.